Xamarin.Essentials/Contacts/Contacts.tizen.cs

```diff
@@ -36,81 +39,93 @@ static async Task<Contact> PlatformPickContactAsync()
-                            {
-                                var item = record.GetChildRecord(TizenContact.Number, i);
-                                var number = item.Get<string>(TizenNumber.NumberData);
-                                var type = (TizenNumber.Types)item.Get<int>(TizenNumber.Type);
-
-                                phones.Add(new ContactPhone(number, GetContactType(type)));
-                            }
-
-                            contact = new Contact(name, phones, emails, ContactType.Unknown);
-                        }
+                            contact = ToContact(record);
                     }
                 }
-
                 tcs.TrySetResult(contact);
             });
-
             return await tcs.Task;
         }
 
-        static ContactType GetContactType(TizenEmail.Types emailType)
+        static Task<IEnumerable<Contact>> PlatformGetAllAsync(CancellationToken cancellationToken)
+        {
+            var contactsList = manager.Database.GetAll(TizenContact.Uri, 0, 0);
+            return Task.FromResult(GetEnumerable());
+
+            IEnumerable<Contact> GetEnumerable()
+            {
+                for (var i = 0; i < contactsList?.Count; i++)
+                {
+                    yield return ToContact(contactsList.GetCurrentRecord());
+                    contactsList.MoveNext();
+                }
+            }
+        }
+
+        static Contact ToContact(ContactsRecord contactsRecord)
+        {
+            var record = contactsRecord.GetChildRecord(TizenContact.Name, 0);
+
+            var phones = new List<ContactPhone>();
+            var phonesCount = contactsRecord.GetChildRecordCount(TizenContact.Number);
+            for (var i = 0; i < phonesCount; i++)
+            {
+                var nameRecord = contactsRecord.GetChildRecord(TizenContact.Number, i);
+                var number = nameRecord.Get<string>(TizenNumber.NumberData);
+                var type = (TizenNumber.Types)nameRecord.Get<int>(TizenNumber.Type);
+
+                phones.Add(new ContactPhone(number, GetContactType(type)));
+            }
+
+            var emails = new List<ContactEmail>();
+            var emailCount = contactsRecord.GetChildRecordCount(TizenContact.Email);
+            for (var i = 0; i < emailCount; i++)
+            {
+                var emailRecord = contactsRecord.GetChildRecord(TizenContact.Email, i);
+                var addr = emailRecord.Get<string>(TizenEmail.Address);
+                var type = (TizenEmail.Types)emailRecord.Get<int>(TizenEmail.Type);
+
+                emails.Add(new ContactEmail(addr, GetContactType(type)));
+            }
+
+            return new Contact(
+                    record?.Get<string>(TizenName.ContactId),
+                    record?.Get<string>(TizenName.Prefix),
+                    record?.Get<string>(TizenName.First),
+                    record?.Get<string>(TizenName.Addition),
```

Comment:
Right, `TizenName.Addition` is the middle name.

Reply:
`TizenName.PhoneticMiddle` and  `TizenName.Addition`  threw me off my train of thought